### ISSUE TITLE
Allow the SonarQube upgrades

### ISSUE THREAD

**Issue by niluid** [lrk/ansible-role-sonarqube#30]
Hi. Thank you for great role. It was useful to install the SonarQube.

It would be nice to have the possibility proceed with SonarQube upgrades. Seems it's not supported at the moment- if we'll execute the role with new version over existing server, the currently running version of app will not be stopped before the starting of new version.

I can make a PR for this, but want to ask if it needed for anybody else. And if needed - discuss the details of implementation.

By current behavior Ansible checking if server have a directory with required version of SonarQube. If not - it starting download of required version.

We probably may add additional check to verify if Sonar service exist (this mean that we probably have the different version installed) - and stop this service after successful download of new binaries. Below example verified with CentOS 7:
```
- name: "Populate service facts to check if previous version installed"
  ansible.builtin.service_facts:

- name: "Stop service before upgrade to {{ sonar_version }}"
  systemd:
    name: sonar
    state: stopped
  when: '"sonar.service" in ansible_facts.services'`
```
Then current version of role update https://github.com/lrk/ansible-role-sonarqube/blob/master/templates/sonar.service.j2 with path to new binaries - but systemd-based OS role will start the old binaries, because the list of handlers doesn't contain the instruction for `systemctl daemon-reload`

By the way, I'm afraid, that daemon-reload might be expected behavior for other changes in https://github.com/lrk/ansible-role-sonarqube/blob/master/templates/sonar.service.j2
 
To make it working I see next options:

1. Replace the `service` module in handlers with `systemd` and update this task with option `daemon_reload: yes` as shown in https://docs.ansible.com/ansible/latest/collections/ansible/builtin/systemd_module.html. If I not wrong, role currently tested with only systemd-based OS - so this change shouldn't change existing compatibility, but will break compatibility with any non-systemd OS for a future.

2.Add additional handler - with name, for example, `systemd daemon-reload` which will be executed only on systemd OSs before the `restart SonarQube service`

3. Don't change the list of role handlers and keep existing `service` module. This could be implemented if /etc/systemd/system/sonar.service will keep not the actual path to SonarQube binaries - but the path to symlink. Imagine we have a symlink /opt/sonarqube/sonarqube-latest which point to directory with the actual version - so sonar.service always contain a path to /opt/sonarqube/sonarqube-latest. Unfortunately last method doesn't take care about 'systemctl daemon-reload' when role made other changes to sonar.service file.

Maybe I missed something in proposed options. Would be appreciate for your feedback.

If you ok with implementing upgrade functionality I want to ask if PR should contain the code for comparing the versions and preventing downgrades. Application probably doesn't support downgrades but somebody may force new version by mistake. I can add some pre-check to prevent such mistakes, but maybe it's better do not add the additional code without strong need.

**Comment by lrk** [comment]
Hi @niluid, thank you for your feedback.

I agree with you, the role should be able to upgrade it's own installation of sonar.
but what if the previous sonar was not installed by this role ? (not same install directory, ...) I don't have the answer.

For the service restart and reloading part, we could create OS specific scripts and trigger them with conditions on ansible os family or maybe the "ansible_service_mgr" ansible fact if it is reliable.

Btw we have to register service file modifications to reload it only when needed. 


Feel free to make a PR with your suggestions.

And if you add the corresponding molecule test cases, it would be awesome.

Thx

**Comment by lrk** [comment]
solved in #33
